### ISSUE TITLE
[BUG] document_importer error : "No version.json file located" and "Database import failed"

### ISSUE THREAD

**Issue by Elaws** [paperless-ngx/paperless-ngx#4115]
### Description

A year ago, I was using paperless-ng (so 1.5.0 supposedly) and regularly made backups using : 

```docker-compose exec -T webserver document_exporter -d ../export```

Today, I installed paperless-ngx again and tried to import the content of this export folder using :

```docker-compose exec -T webserver document_importer ../export```

However, I got the error logs attached below.

Is there a way to import my documents in paperless-ngx ? 

Thank you.

### Steps to reproduce

Cf. bug description.

### Webserver logs

```bash
No version.json file located
Checking the manifest
Database import failed
No version information present
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/models/options.py", line 669, in get_field
    return self.fields_map[field_name]
KeyError: 'user'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/json.py", line 70, in Deserializer
    yield from PythonDeserializer(objects, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/python.py", line 130, in Deserializer
    field = Model._meta.get_field(field_name)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/options.py", line 671, in get_field
    raise FieldDoesNotExist(
django.core.exceptions.FieldDoesNotExist: SavedView has no field named 'user'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/paperless/src/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 440, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 402, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/usr/src/paperless/src/documents/management/commands/document_importer.py", line 147, in handle
    raise e
  File "/usr/src/paperless/src/documents/management/commands/document_importer.py", line 128, in handle
    call_command("loaddata", manifest_path)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 198, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 448, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 102, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 163, in loaddata
    self.load_label(fixture_label)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 251, in load_label
    for obj in objects:
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/json.py", line 74, in Deserializer
    raise DeserializationError() from exc
django.core.serializers.base.DeserializationError: Problem installing fixture '/usr/src/paperless/export/manifest.json':
```

### Paperless-ngx version

1.17.4

### Host OS

Windows 11

### Installation method

Docker - official image

### Browser

Firefox

**Comment by stumpylog** [comment]
Imports are highly unlike to work across versions.  You will need to start from your previous version, import, then upgrade in place.  In a newer version, there's a warning when attempting this.

**Comment by Elaws** [comment]
@stumpylog : Thank you. What would be the best way to achieve this ?

Should I simply switch back to `image:jonaswinkler/paperless-ng:latest` in the `docker_compose.yml`, then do a `docker-compose pull`, do the import, and then do another pull from the paperless-ngx repo ? Will it keep my imported documents doing so ?

**Comment by shamoon** [comment]
Yes https://docs.paperless-ngx.com/setup/#migrating-from-paperless-ng

**Comment by stumpylog** [comment]
If you have the existing installation still, you can (in theory) change the image from `jonaswinkler/paperless-ng:latest` to `ghcr.io/paperless-ngx/paperless-ngx:latest` and it should work.  

Of course, that's many large updates at once, so you could go version by version (`ghcr.io/paperless-ngx/paperless-ngx:1.5.0`, then `ghcr.io/paperless-ngx/paperless-ngx:1.6.0`, etc).  It's time consuming, but may give more peace of mind.

In either case, I would be sure to keep a backup of folders, databases, etc, just since it's so massive a jump.  (But also a good one in my opinion)

**Comment by Elaws** [comment]
So I just installed paperless-ng 1.5.0, server is up running, and when I import my "export" folder, I have the following error :

```
PS D:\paperless-ng> docker-compose exec -T webserver document_importer ../export
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedColumn: column "user_id" of relation "documents_savedview" does not exist
LINE 1: UPDATE "documents_savedview" SET "user_id" = 2, "name" = 'Ét...
                                         ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/paperless/src/manage.py", line 11, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/src/paperless/src/documents/management/commands/document_importer.py", line 77, in handle
    call_command("loaddata", manifest_path)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 181, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 78, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 123, in loaddata
    self.load_label(fixture_label)
  File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/loaddata.py", line 190, in load_label
    obj.save(using=self.using)
  File "/usr/local/lib/python3.9/site-packages/django/core/serializers/base.py", line 223, in save
    models.Model.save_base(self.object, using=using, raw=True, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 763, in save_base
    updated = self._save_table(
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 845, in _save_table
    updated = self._do_update(base_qs, using, pk_val, values, update_fields,
  File "/usr/local/lib/python3.9/site-packages/django/db/models/base.py", line 899, in _do_update
    return filtered._update(values) > 0
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 802, in _update
    return query.get_compiler(self.db).execute_sql(CURSOR)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1559, in execute_sql
    cursor = super().execute_sql(result_type)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: Problem installing fixture '/usr/src/paperless/src/../export/manifest.json': Could not load documents.SavedView(pk=2): column "user_id" of relation "documents_savedview" does not exist
LINE 1: UPDATE "documents_savedview" SET "user_id" = 2, "name" = 'Ét...
```

Any idea how to fix this ?

Thanks a lot.

**Comment by stumpylog** [comment]
Not for sure, but make sure you're starting with no database as well.  If you started 1.17, it will have done commands to the db

**Comment by Elaws** [comment]
> Not for sure, but make sure you're starting with no database as well.  If you started 1.17, it will have done commands to the db

Thanks : I created another folder for paperless-ng and did the dock-compose pull there (with paperless-ng repo in the .yml). 

Since I created a new folder, I suppose Docker created a new paperless-ng image that has nothing to do with my other paperless-ngx folder, right ? Or should have I uninstalled paperless-ngx before ?

**Comment by stumpylog** [comment]
You're using Postgres, so a new folder where `docker-compose.yml` lives would not create a new data base.  You would either need a new container, different DB name or to use a new volume, all depending on how you've set up your installation

**Comment by Elaws** [comment]
> You're using Postgres, so a new folder where `docker-compose.yml` lives would not create a new data base.  You would either need a new container, different DB name or to use a new volume, all depending on how you've set up your installation

Thank you. So how could I just uninstall paperless-ngx to have a clean install of paperless-ng, properly import my documents, and finally upgrade to paperless-ngx ?

I'm just using Docker for Windows and the `docker-compose pull` (or `up`). I've 2 folders : One for paperless-ngx and another for paperless-ng.

**Comment by stumpylog** [comment]
Perhaps for simplicity then, you should just use SQLite, the default database. That's clear that if the data folder is empty, there's no database.  Using Postgres is more complicated and if you're not familiar with Docker, it might be easy to share accidentally.

**Comment by Elaws** [comment]
So I deleted the volumes and images in Docker and used the old `docker-compose.yml` I had a year ago. Then `docker-compose up` from `jonaswinkler/paperless-ng:latest` and I was finally able to import my old database.

After that, I did a `docker-compose up` from `ghcr.io/paperless-ngx/paperless-ngx:latest` with a new `docker-compose.yml`.

I get a fatal error : 

`The database directory was initialized by PostgreSQL 13, which is not compatible with version 15.4`.

Please find below the old and new `docker-compose.yml` I used : 

---
OLD (used with paperless-ng) : 

```
version: "3.4"
services:
  broker:
    image: redis:6.0
    restart: unless-stopped

  db:
    image: postgres:13
    restart: unless-stopped
    volumes:
      - pgdata:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: paperless
      POSTGRES_USER: paperless
      POSTGRES_PASSWORD: paperless

  webserver:
    image: jonaswinkler/paperless-ng:latest
    restart: unless-stopped
    depends_on:
      - db
      - broker
      - gotenberg
      - tika
    ports:
      - 8000:8000
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - data:/usr/src/paperless/data
      - media:/usr/src/paperless/media
      - ./export:/usr/src/paperless/export
      - D:\paperless-ng\data:/usr/src/paperless/consume
    env_file: docker-compose.env
    environment:
      PAPERLESS_REDIS: redis://broker:6379
      PAPERLESS_DBHOST: db
      PAPERLESS_TIKA_ENABLED: 1
      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
      PAPERLESS_TIKA_ENDPOINT: http://tika:9998

  gotenberg:
    image: thecodingmachine/gotenberg
    restart: unless-stopped
    environment:
      DISABLE_GOOGLE_CHROME: 1

  tika:
    image: apache/tika
    restart: unless-stopped

volumes:
  data:
  media:
  pgdata:
```
---
NEW (used to update to paperless-ngx) :

```
version: "3.4"
services:
  broker:
    image: docker.io/library/redis:7
    restart: unless-stopped
    volumes:
      - redisdata:/data

  db:
    image: docker.io/library/postgres:15
    restart: unless-stopped
    volumes:
      - pgdata:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: paperless
      POSTGRES_USER: paperless
      POSTGRES_PASSWORD: paperless

  webserver:
    image: ghcr.io/paperless-ngx/paperless-ngx:latest
    restart: unless-stopped
    depends_on:
      - db
      - broker
      - gotenberg
      - tika
    ports:
      - "8000:8000"
    healthcheck:
      test: ["CMD", "curl", "-fs", "-S", "--max-time", "2", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - data:/usr/src/paperless/data
      - media:/usr/src/paperless/media
      - ./export:/usr/src/paperless/export
      - D:\paperless-ng\data:/usr/src/paperless/consume
    env_file: docker-compose.env
    environment:
      PAPERLESS_REDIS: redis://broker:6379
      PAPERLESS_DBHOST: db
      PAPERLESS_TIKA_ENABLED: 1
      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
      PAPERLESS_TIKA_ENDPOINT: http://tika:9998

  gotenberg:
    image: docker.io/gotenberg/gotenberg:7.8
    restart: unless-stopped

    # The gotenberg chromium route is used to convert .eml files. We do not
    # want to allow external content like tracking pixels or even javascript.
    command:
      - "gotenberg"
      - "--chromium-disable-javascript=true"
      - "--chromium-allow-list=file:///tmp/.*"

  tika:
    image: ghcr.io/paperless-ngx/tika:latest
    restart: unless-stopped

volumes:
  data:
  media:
  pgdata:
  redisdata:
```

Thanks for your help.

**Comment by Elaws** [comment]
@stumpylog : Please see my comment above, and :

> Perhaps for simplicity then, you should just use SQLite, the default database. That's clear that if the data folder is empty, there's no database. Using Postgres is more complicated and if you're not familiar with Docker, it might be easy to share accidentally.

I think I followed the recommendations on the official documentation : 

> For new installations, it is recommended to use PostgreSQL as the database backend.

(Source : https://docs.paperless-ngx.com/setup/)

I don't remember what was the recommendation with paperless-ng back then, but I probably followed the use documentation too.

My old database is on PostgreSQL and I unfortunately can't start from scratch on a new database.

**Comment by stumpylog** [comment]
> docker.io/library/postgres:15

You can easily downgrade Postgres by changing that tag

**Comment by Elaws** [comment]
**I solved it !**

1. Follow steps described in [this comment above](https://github.com/paperless-ngx/paperless-ngx/issues/4115#issuecomment-1704355166), but use *new* `docker-compose.yml` when installing paperless-ng, with `ghcr.io/paperless-ngx/paperless-ngx:latest` replaced by `jonaswinkler/paperless-ng:latest`.

Then, the document import still worked properly (I thought it wouldn't work without the old `docker-compose.yml`). 

After that, I just replaced `jonaswinkler/paperless-ng:latest` by `ghcr.io/paperless-ngx/paperless-ngx:latest` in the `docker-compose.yml` and can still see my documents in the latest paperless-ngx version.
